### PR TITLE
55: set up default values and remove with()

### PR DESCRIPTION
**Pull request by jshvarts**
val note = Note()
        Timber.d("note: " + note)
        Timber.d("empty string: " + Note.EMPTY_STRING)
        Timber.d("empty timestamp: " + Note.EMPTY_TIMESTAMP)
        Timber.d("copy: " + note.copy(id = "10", title = "new 10"))

prints:

note: Note(id=, title=, body=, created=0, updated=0)
empty string: 
empty timestamp: 0
copy: Note(id=10, title=new 10, body=, created=0, updated=0)

### CODE REVIEW

File: app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt [hunk 1]
```diff
@@ -3,20 +3,14 @@ package com.exallium.todoapp.entities
...
-             created: Long? = null,
-             updated: Long? = null) = Note(
-            id ?: this.id,
-            title ?: this.title,
-            body ?: this.body,
-            created ?: this.created,
-            updated ?: this.updated)
+    companion object {
+        val EMPTY_STRING = ""
+        val EMPTY_TIMESTAMP = 0L
```
Comment (exallium) [inline app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt]: Technically 0L is a valid timestamp.  let's make this -1L

File: app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt [hunk 2]
```diff
@@ -3,20 +3,14 @@ package com.exallium.todoapp.entities
...
-                val body: String,
-                val created: Long,
-                val updated: Long) {
+data class Note(val id: String = EMPTY_STRING,
+                val title: String = EMPTY_STRING,
+                val body: String = EMPTY_STRING,
+                val created: Long = EMPTY_TIMESTAMP,
+                val updated: Long = EMPTY_TIMESTAMP) {
 
-    fun with(id: String? = null,
```
Comment (exallium) [inline app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt:12]: =D

**Review by exallium — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt (hunk 1 above)
Comment (jshvarts) [reply app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt]: done

Comment (jshvarts) [reply app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt]: so the current validation i have checks if the field is not just empty but blank which is better than comparing to Note.EMPTY_STRING. unless i add several specific validation methods on Note all checking the same thing (empty string), my current validation is fine and the empty fields in the constructor are only used for something like firebase that requires no args constructor.

**Pull request closed without merging**